libs/src/oracle-sdk-v2/services/oraclev1/gql/queries.ts

```diff
@@ -12,13 +13,25 @@ import request, { gql } from "graphql-request";
 export async function getPriceRequests(
   url: string,
   chainId: ChainId,
-  oracleType: OracleType
+  oracleType: OracleType,
+  addErrorMessage: (message: ErrorMessage) => void
 ) {
-  const chainName = chainsById[chainId];
-  const queryName = makeQueryName(oracleType, chainName);
-  const isV2 = oracleType === "Optimistic Oracle V2";
-  const result = await fetchAllRequests(url, queryName, isV2);
-  return result;
+  try {
+    const chainName = chainsById[chainId];
+    const queryName = makeQueryName(oracleType, chainName);
+    const isV2 = oracleType === "Optimistic Oracle V2";
+    const result = await fetchAllRequests(url, queryName, isV2);
+    return result;
+  } catch (e) {
+    addErrorMessage({
+      text: "TheGraph is experiencing downtime. Please use the legacy dapp while they rectify the issue",
+      link: {
+        text: "Legacy Dapp",
```

Comment:
Probably would be better to just hyperlink the first "legacy dapp" in the message rather than repeating.